<?xml version="1.0"?>
<launch>
  <arg name="world_name" default="field" />
  <arg name="name" default="costar_husky_sensor_config_1" />
  <arg name="link_name" value="base_link" />
  <arg name="sensor_prefix" value="/world/$(arg world_name)/model/$(arg name)/link/$(arg link_name)/stereo_camera" />
  <arg name="sensor_prefix_depth" value="/world/$(arg world_name)/model/$(arg name)/link/$(arg link_name)/sensor/realsense_d435" />

  <node
    pkg="ros_ign_bridge"
    type="parameter_bridge"
    name="ros_ign_bridge_pose"
    args="/clock@rosgraph_msgs/Clock[ignition.msgs.Clock"></node>

  <group ns="$(arg name)">
    <!--Create multipe bridges so that it can run in parallel-->
    <node
      pkg="ros_ign_bridge"
      type="parameter_bridge"
      name="ros_ign_bridge_pose"
      args="/model/$(arg name)/pose@tf2_msgs/TFMessage[ignition.msgs.Pose_V">
      <remap from="/model/$(arg name)/pose" to="pose" />
    </node>
    <node
      pkg="ros_ign_bridge"
      type="parameter_bridge"
      name="ros_ign_bridge_pose_static"
      args="/model/$(arg name)/pose_static@tf2_msgs/TFMessage[ignition.msgs.Pose_V">
      <remap from="/model/$(arg name)/pose_static" to="pose_static" />
    </node>

    <node
      pkg="ros_ign_bridge"
      type="parameter_bridge"
      name="ros_ign_bridge_twist"
      args="/cmd_vel@geometry_msgs/Twist@ignition.msgs.Twist">
      <remap from="/cmd_vel" to="cmd_vel" />
    </node>

    <node
      pkg="ros_ign_bridge"
      type="parameter_bridge"
      name="ros_ign_bridge_odom"
      args="/model/$(arg name)/odometry@nav_msgs/Odometry[ignition.msgs.Odometry">
      <remap from="/model/$(arg name)/odometry" to="odom" />
    </node>

    <!--Stereo RGB camera -->
    <node
      pkg="ros_ign_bridge"
      type="parameter_bridge"
      name="ros_ign_bridge_camera_left"
      args="$(arg sensor_prefix)/left/camera_info@sensor_msgs/CameraInfo[ignition.msgs.CameraInfo">
      <remap from="$(arg sensor_prefix)/left/camera_info" to="left/camera_info" />
    </node>
    <node
      pkg="ros_ign_bridge"
      type="parameter_bridge"
      name="ros_ign_bridge_camera_medio"
      args="$(arg sensor_prefix_depth)/camera_info@sensor_msgs/CameraInfo[ignition.msgs.CameraInfo">
      <remap from="$(arg sensor_prefix_depth)/camera_info" to="medio/camera_info" />
    </node>

    <node
      pkg="ros_ign_image"
      type="image_bridge"
      name="ros_ign_image_camera_left"
      args="$(arg sensor_prefix)/left/image_raw"
    >
      <remap from="$(arg sensor_prefix)/left/image_raw" to="left/image_raw" />
    </node>

    <node
      pkg="ros_ign_bridge"
      type="parameter_bridge"
      name="ros_ign_bridge_camera_right"
      args="$(arg sensor_prefix)/right/camera_info@sensor_msgs/CameraInfo[ignition.msgs.CameraInfo">
      <remap from="$(arg sensor_prefix)/right/camera_info" to="right/camera_info" />
    </node>
    <node
      pkg="ros_ign_image"
      type="image_bridge"
      name="ros_ign_image_camera_right"
      args="$(arg sensor_prefix)/right/image_raw"
    >
      <remap from="$(arg sensor_prefix)/right/image_raw" to="right/image_raw" />
    </node>

    <node
      pkg="ros_ign_image"
      type="image_bridge"
      name="ros_ign_image_camera_medio"
      args="$(arg sensor_prefix_depth)/image"
    >
      <remap from="$(arg sensor_prefix_depth)/image" to="medio/image_raw" />
    </node>

    <!--Depth camera -->
    <node
      pkg="ros_ign_image"
      type="image_bridge"
      name="ros_ign_depth_image_camera_medio"
      args="$(arg sensor_prefix_depth)/depth_image"
    >
      <remap from="$(arg sensor_prefix_depth)/depth_image" to="medio/depth_image" />
    </node>

  </group>
</launch>